### DISCUSSION TITLE
Extend the capabilities of extract_text()

### DISCUSSION THREAD

**Discussion by erenirmak** [py-pdf/pypdf#1502] [Ideas]
Some PDF files might be scanned documents, consist of images instead of texts. Some PDF documents might have images beside texts. For both situations, we lose some information.

extract_text() function of PyPDF2 can be extended to process the images automatically as well as usual texts. It would make our life easier. Though I don't know the backend. Is that possible to implement?

**Comment by MartinThoma** [comment]
https://pypdf2.readthedocs.io/en/latest/user/extract-text.html

> Sometimes PDFs do not contain the text as it’s displayed, but instead an image. You notice that when you cannot copy the text. Then there are PDF files that contain an image and a text layer in the background. That typically happens when a document was scanned. Although the scanning software (OCR) is pretty good today, it still fails once in a while. PyPDF2 is no OCR software; it will not be able to detect those failures. PyPDF2 will also never be able to extract text from images.
